Azure/sonic-buildimage
# Add default catch block in portsyncd

**- What I did**
Added a default catch block in portsyncd code as current catch block only accepts parameter of type Exception. In the throw currently, a const char * is passed which leads to a core dumped scenario.

**- How I did it**
Added a default catch bl;ock.

**- How to verify it**
To test, Build a sonic-Broadcom image. 
Loaded the image in a DUT, rebooted the DUT without a minigraph.xml or config_db.json in /etc/sonic. 
While booting up, when systemd tries to start portsyncd process, it passes the parameter port_config.ini. The right port_config.ini is selected and placed in the right directory in swss docker before starting portsyncd process. The right port_config.ini is selected based HW SKU and platform type which is obtained from DEVICE_METADATA. If the device configuration is not loaded correctly, then DEVICE_METADATA will not be read, and port_config.ini is not picked up. With this, if portsyncd is started without the right port_config.ini, the portsyncd process will throw and exception. 

Before the fix error looked like:

<14>Aug  5 22:20:40 MWH01-0101-0334-13T0 supervisord: portsyncd terminate called after throwing an instance of 'char const*'
<30>Aug  5 22:20:40 MWH01-0101-0334-13T0 swss.sh[1901]: 2019-08-05 22:20:40,806 INFO exited: portsyncd (terminated by SIGABRT (core dumped); not expected)

After fix the error will look like:
Aug 14 23:24:55.845720 sonic INFO supervisord: portsyncd Exception had been thrown in deamon

**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**
